File: frontend/lib/services/statistics_service.dart
import 'api_service.dart';
import '../models/achievement.dart';
import '../models/statistics.dart';

class StatisticsService {
  final ApiService _api = ApiService();

  Future<DailyStats> getDailyStats() async {
    final resp = await _api.dio.get('/api/statistics/daily');
    return DailyStats.fromJson(resp.data['data']);
  }

  Future<WeeklyStats> getWeeklyStats() async {
    final resp = await _api.dio.get('/api/statistics/weekly');
    return WeeklyStats.fromJson(resp.data['data']);
  }

  Future<Trend> getTrend(int days) async {
    final resp = await _api.dio.get('/api/statistics/trend', queryParameters: {'days': days});
    return Trend.fromJson(resp.data['data']);
  }

  Future<void> reportDuration(int seconds) async {
    await _api.dio.post('/api/statistics/duration', data: {'seconds': seconds});
  }

  Future<List<DailyStats>> getHeatmap({int year = 2026, int? month}) async {
    final params = <String, dynamic>{'year': year};
    if (month != null) params['month'] = month;
    final resp = await _api.dio.get('/api/statistics/heatmap', queryParameters: params);
    return (resp.data['data'] as List).map((e) => DailyStats.fromJson(e)).toList();
  }

  Future<SummaryData> getSummary() async {
    final resp = await _api.dio.get('/api/statistics/summary');
    return SummaryData.fromJson(resp.data['data']);
  }

  Future<List<CategoryRetention>> getRetention() async {
    final resp = await _api.dio.get('/api/statistics/retention');
    return (resp.data['data'] as List)
        .map((e) => CategoryRetention.fromJson(e))
        .toList();
  }
}

File: backend/src/main/java/com/mindflow/service/StatisticsService.java
package com.mindflow.service;

import com.mindflow.vo.CategoryRetentionVO;
import com.mindflow.vo.DailyStatsVO;
import com.mindflow.vo.SummaryVO;
import com.mindflow.vo.TrendVO;
import com.mindflow.vo.WeeklyStatsVO;

import java.util.List;
import java.util.Map;

/**
 * 学习统计服务接口
 */
public interface StatisticsService {

    /** 增加笔记计数 */
    void incrementNoteCount();

    /** 增加复习计数 */
    void incrementReviewCount();

    /** 增加面试题计数 */
    void incrementInterviewCount();

    /** 增加学习时长 */
    void addStudyDuration(int seconds);

    /** 今日统计 */
    DailyStatsVO getDailyStats();

    /** 本周统计 */
    WeeklyStatsVO getWeeklyStats();

    /** 学习趋势（近30天） */
    TrendVO getTrend(int days);

    /** 学习热力图 */
    List<DailyStatsVO> getHeatmap(int year, Integer month);

    /** 学习成就摘要 */
    SummaryVO getSummary();

    /** 知识保留度（按分类） */
    List<CategoryRetentionVO> getRetention();
}